lib/src/local_working_copy.rs

```diff
@@ -939,11 +933,12 @@ impl TreeState {
             },
             #[cfg(not(feature = "watchman"))]
             Some(FsmonitorKind::Watchman) => {
-                return Err(SnapshotError::FsmonitorError(
-                    "Cannot query Watchman because jj was not compiled with the `watchman` \
-                     feature (consider disabling `core.fsmonitor`)"
-                        .to_string(),
-                ));
+                return Err(SnapshotError::Other {
+                    message: "Failed to query the filesystem monitor".to_string(),
+                    err: "Cannot query Watchman because jj was not compiled with the `watchman` \
+                          feature (consider disabling `core.fsmonitor`)"
+                        .into(),
+                });
```

Comment:
nit: another way is to remove `message` field from `Other(Box<dyn Error ..>)` and attach it to the inner error object as needed. I don't know which is common, but sometimes the inner error has enough context (or it is just a string), and the outer `message` can be redundant.

Reply:
True. I feel like the message might be useful here since we pass different messages. I'd rather leave it for now.